app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/WifiDirectManager.kt

```diff
@@ -243,8 +243,19 @@ class WifiDirectManager @Inject constructor(
     if (BuildConfig.DEBUG) {
       Log.d(TAG, "Starting handshake")
     }
-    peerGroupHandshakeAsyncTask = PeerGroupHandshakeAsyncTask(this)
-      .also { it.execute() }
+    peerGroupHandshake = PeerGroupHandshake(this)
+      .also {
```

Comment:
I'd stop using this `also` paradigm in this file, it was fine when it was 1 statement but that is no longer the case

Reply:
and all task variables can be local because we no longer need a reference to cancel later on

Reply:
done

Reply:
@gouri-panda you may have missed the 2nd comment here

Reply:
D'oh! right on.

Reply:
done now